packages/contracts/contracts/minter-suite/Minters/MinterSetPriceHolderV5.sol

```diff
@@ -0,0 +1,790 @@
+ * Additional admin and artist privileged roles may be described on other
+ * contracts that this minter integrates with.
+ * ----------------------------------------------------------------------------
+ * This contract allows gated minting with support for vaults to delegate minting
+ * privileges via an external delegation registry. This means a vault holding an
+ * allowed token can delegate minting privileges to a wallet that is not holding an
+ * allowed token, enabling the vault to remain air-gapped while still allowing minting.
+ * The delegation registry contract is responsible for managing these delegations,
+ * and is available at the address returned by the public immutable
+ * `delegationRegistryAddress`. At the time of writing, the delegation
+ * registry enables easy delegation configuring at https://delegate.cash/.
+ * Art Blocks does not guarentee the security of the delegation registry, and
+ * users should take care to ensure that the delegation registry is secure.
+ * Delegations must be configured by the vault owner prior to purchase. Supported
+ * delegation types include token-level, contract-level (via genArt721CoreAddress), or
+ * wallet-level delegation. Contract-level delegations must be configured for the core
+ * token contract as returned by the public immutable variable `genArt721CoreAddress`.
+ */
+contract MinterSetPriceHolderV5 is
+    ReentrancyGuard,
+    ISharedMinterV0,
+    IFilteredSharedHolder
+{
+    // add Enumerable Set methods
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    // Minter filter address this minter interacts with
+    address public immutable minterFilterAddress;
+
+    // Minter filter this minter may interact with.
+    IMinterFilterV1 private immutable minterFilter;
+
+    // Delegation registry address
+    address public immutable delegationRegistryAddress;
+
+    // Delegation registry address
+    IDelegationRegistry private immutable delegationRegistryContract;
+
+    // minterType for this minter
+    string public constant minterType = "MinterSetPriceHolderV5";
+
+    // minter version for this minter
+    string public constant minterVersion = "v5.0.0";
+
+    uint256 constant ONE_MILLION = 1_000_000;
+
+    // contractAddress => projectId => base project config
+    mapping(address => mapping(uint256 => ProjectConfig))
+        private _projectConfigMapping;
+
+    /**
+     * coreContract => projectId => ownedNFTAddress => ownedNFTProjectIds => bool
+     * projects whose holders are allowed to purchase a token on `projectId`
+     */
+    mapping(address => mapping(uint256 => mapping(address => mapping(uint256 => bool))))
+        private _allowedProjectHoldersMapping;
+
+    // STATE VARIABLES FOR SplitFundsLib begin here
+    // contractAddress => IsEngineCache
+    mapping(address => SplitFundsLib.IsEngineCache) private _isEngineCaches;
+    // STATE VARIABLES FOR SplitFundsLib end here
+
+    // STATE VARIABLES FOR MaxInvocationsLib begin here
```

Comment:
```suggestion
    ////////////////////////////////////////////////////////////////////////////////////////////////////////////
    // STATE VARIABLES FOR MaxInvocationsLib begin here
    ////////////////////////////////////////////////////////////////////////////////////////////////////////////
```

Reply:
Nit suggestion to do something to this effect here and throughout this PR to improve readability of these blocks.